torchtext/datasets/sogounews.py

```diff
@@ -35,10 +38,17 @@
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=5)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
-def SogouNews(root, split):
-    path = _download_extract_validate(root, URL, MD5, os.path.join(root, _PATH), os.path.join(root, _EXTRACTED_FILES[split]),
-                                      _EXTRACTED_FILES_MD5[split], hash_type="md5")
-    logging.info('Creating {} data'.format(split))
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+@_wrap_split_argument(("train", "test"))
+def SogouNews(root: str, split: Union[Tuple[str], str]):
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL])
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
+    )
+    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_dp = FileOpener(cache_dp, mode="b")
+    extracted_files = cache_dp.read_from_tar()
+    filter_extracted_files = extracted_files.filter(lambda x: split + ".csv" in x[0])
```

Comment:
NIT: Let's be consistent with how we're doing the check for the `split` file existing in the folder.
We could do `lambda x: _EXTRACTED_FILES[split] in x[0]` which is what you did in [this PR](https://github.com/pytorch/text/pull/1499/files#diff-61c4706569928ac5352feafe9003545f246296ecc93c10e20120747f3707ccbdR57)